controllers/helmchartinstall_controller.go

```diff
@@ -77,61 +126,255 @@ func (r *HelmChartInstallReconciler) render(ctx context.Context, claim *helmv1al
 	// - Stack configuration will be coming from a Kubernetes object, probably the
 	//   Stack itself
-	//stackConfiguration := ''
-	return unstructured.Unstructured{}, nil
+	namespacedName, err := client.ObjectKeyFromObject(claim)
+
+	if err != nil {
+		r.Log.V(0).Info("getStackConfiguration returning early because of error creating object key", "err", err, "claim", claim)
+		return nil, err
+	}
+
+	configuration := &v1alpha1.StackConfiguration{}
+	if err := r.Client.Get(ctx, namespacedName, configuration); err != nil {
+		r.Log.V(0).Info("getStackConfiguration returning early because of error fetching configuration", "err", err, "claim", claim)
+		return nil, err
+	}
+
+	r.Log.V(0).Info("getStackConfiguration returning configuration", "configuration", configuration)
+	return configuration, nil
+}
+
+/**
+ * When a behavior is triggered, we want to know which behavior exactly we are executing.
+ *
+ * In most cases, this will probably be configured ahead of time by the setup controller, rather
+ * than being fetched at runtime by the render controller.
+ */
+func (r *HelmChartInstallReconciler) getBehavior(
+	ctx context.Context,
+	claim *unstructured.Unstructured,
+	configuration *v1alpha1.StackConfiguration,
+) (*v1alpha1.StackConfigurationBehavior, error) {
+	targetResourceGroupVersion, targetResourceKind := claim.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
+	targetGroupKindVersion := fmt.Sprintf("%s.%s", targetResourceKind, targetResourceGroupVersion)
+
+	var targetResourceBehavior v1alpha1.StackConfigurationBehavior
+	// TODO handle missing keys gracefully
+	targetResourceBehavior, ok := configuration.Spec.Behaviors.CRDs[targetGroupKindVersion]
+
+	if !ok {
+		// TODO error condition with a real error returned
+		r.Log.V(0).Info("Couldn't find a configured behavior!",
+			"claim", claim,
+			"configuration", configuration,
+			"targetGroupKindVersion", targetGroupKindVersion,
+		)
+		return nil, nil
+
+	}
+
+	if len(targetResourceBehavior.Resources) == 0 {
+		// TODO error condition with a real error returned
+		// TODO it'd be nice to enforce this on acceptance or creation if possible
+		r.Log.V(0).Info("Couldn't find resources for configured behavior!", "claim", claim, "configuration", configuration)
+		return nil, nil
+
+	}
+
+	return &targetResourceBehavior, nil
+}
+
+/**
+ * When a behavior executes, the resource engine is configured by the
+ * object which triggered the behavior. This method encapsulates the logic to
+ * create the resource engine configuration from the object's fields.
+ *
+ * TODO Currently creation fails if the config map already exists, but it should
+ * succeed instead.
+ */
+func (r *HelmChartInstallReconciler) createBehaviorEngineConfiguration(
+	ctx context.Context,
+	claim *unstructured.Unstructured,
+	stackConfig *v1alpha1.StackConfiguration,
+) (*corev1.ConfigMap, error) {
+	// yamlyamlyamlyamlyaml
+	// TODO if spec is missing, that won't work very well
+	spec, ok := claim.Object["spec"]
```

Comment:
`spec` seems like a good candidate for a constant.

Reply:
This is now a constant